server/server.go

```diff
@@ -791,6 +800,35 @@ func (h *RequestHandler) DeletePostRoute(res http.ResponseWriter, req *http.Requ
 	http.Redirect(res, req, threadURL, http.StatusSeeOther)
 }
 
+func (h *RequestHandler) EditPostRoute(res http.ResponseWriter, req *http.Request) {
+	postid, ok := util.GetURLPortion(req, 3)
+	loggedIn, userid := h.IsLoggedIn(req)
+	post, err := h.db.GetPost(postid)
+
+	if !ok || errors.Is(err, sql.ErrNoRows) {
+		title := h.translator.Translate("ErrEdit404")
+		data := GenericMessageData{
+			Title:   title,
+			Message: h.translator.Translate("ErrEdit404Message"),
+		}
+		h.renderGenericMessage(res, req, data)
+		return
+	}
+	if !loggedIn || userid != post.AuthorID {
+		h.renderGeneric401(res, req)
+		return
+	}
+	if req.Method == "POST" {
+		content := req.PostFormValue("content")
+		h.db.EditPost(content, postid)
+		post.Content = content
+	}
+	// update rss feed
+	h.rssFeed = GenerateRSS(h.db, h.config)
```

Comment:
probably not needed / potentially confusing based on how the feed is generated rn? it basically takes the last post of each thread as the "latest post", linking to it and saying something like "[date] [person] posted"

Reply:
ah ya I didnt look to close on this. I'll change it!